<launch>
  <arg name="usb_port"                          default="/dev/ttyUSB0" />
  <arg name="dxl_baud_rate"                     default="1000000" />
  <arg name="robot_model"                       default="px100"/>
  <arg name="robot_name"                        default="$(arg robot_model)"/>
  <arg name="base_link_frame"                   default="base_link"/>
  <arg name="show_ar_tag"                       default="false"/>
  <arg name="use_world_frame"                   default="true"/>
  <arg name="external_urdf_loc"                 default=""/>
  <arg name="use_rviz"                          default="false"/>
  <arg name="mode_configs"                      default="$(find px100_control)/config/modes.yaml"/>
  <arg name="dof"                               default="4"/>
  <arg name="show_gripper_bar"                  default="true"/>
  <arg name="show_gripper_fingers"              default="true"/>
  <arg name="motor_configs"                     default="$(find px100_control)/config/px100.yaml"/>
  <arg name="load_configs"                      default="true" />
  <!-- <param name="dynamixel_info"                  value="$(find px100_control)/config/motor_config_template.yaml"/>   -->
  <rosparam file="$(find px100_control)/config/4dof_controllers.yaml" command="load" ns="$(arg robot_name)"/>
  <rosparam file="$(find px100_control)/config/hardware.yaml" command="load" ns="$(arg robot_name)"/>

  <!-- <include file="$(find dynamixel_workbench_controllers)/launch/dynamixel_controllers.launch">
    <arg name="usb_port" value = "$(arg usb_port)" />
    <arg name="dxl_baud_rate" value = "$(arg dxl_baud_rate)" />
  </include> -->

  <include file="$(find px100_description)/launch/px100_description.launch">
    <arg name="robot_model"                       value="$(arg robot_model)"/>
    <arg name="robot_name"                        value="$(arg robot_name)"/>
    <arg name="base_link_frame"                   value="$(arg base_link_frame)"/>
    <arg name="show_ar_tag"                       value="$(arg show_ar_tag)"/>
    <arg name="show_gripper_bar"                  value="$(arg show_gripper_bar)"/>
    <arg name="show_gripper_fingers"              value="$(arg show_gripper_fingers)"/>
    <arg name="use_world_frame"                   value="$(arg use_world_frame)"/>
    <arg name="external_urdf_loc"                 value="$(arg external_urdf_loc)"/>
    <arg name="use_rviz"                          value="$(arg use_rviz)"/>
  </include>

  <node
    name="controller_spawner"
    pkg="controller_manager"
    type="controller_manager"
    respawn="false"
    output="screen"
    ns="$(arg robot_name)"
    args="spawn arm_controller gripper_controller">
  </node>

  <node 
    name="joint_state_publisher"
    pkg="px100_control"
    type="joint_state_publisher.py">
  </node>

  <node
    name="mr_kinetik_dxl_interface"
    pkg="px100_dynamixel_interface"
    output="screen"
    type="mr_kinetik_dxl_interface"
    ns="$(arg robot_name)">
    <param name="motor_configs"                   value="$(arg motor_configs)"/>
    <param name="mode_configs"                    value="$(arg mode_configs)"/>
    <param name="load_configs"                    value="$(arg load_configs)"/>
  </node>

  <node
    name="mr_kinetik_hardware_interface"
    pkg="px100_control"
    type="mr_kinetik_hardware_interface"
    output="screen"
    ns="$(arg robot_name)">
  </node> 

</launch>

